### PR TITLE
[Bugfix][Model Runner V2][Spec Decode] Decouple the draft's gumbel noise stream from the target's

### PR DESCRIPTION
**Pull request by TheEpicDolphin**
# Context
Picking up the work from #47386. During target verification, the Gumbel noise used to re-sample a rejected draft token is drawn from the same Philox offset as the noise that produced that draft token (the same seed, pos), so byte-identical noise. Conditioned on the proposal winning the argmax, the other tokens' Gumbels are truncated below that max, most tightly for the tokens the draft ranked highest, but lost (high-q loser tokens generally have smaller noise, because otherwise they would have won), so the residual under-weights exactly those tokens and the output distribution is no longer the target's.

This affects `draft_sample_method="probabilistic"` only. Under the default greedy, `draft_logits` is None, the draft never calls `gumbel_sample`, and there is no shared noise vector.

# This PR
Adds a salt (1 << 30) to the offset of the draft's Philox stream, decoupling the draft's Gumbel noise from the target's. Verification is a probability-ratio test against the cached draft distribution rather than a Gumbel coupling, so nothing depends on the two streams matching.

Also refactors how the speculators compute the position used as the sampling-stream index, so each site states its own convention instead of relying on an implicit +1 inside sample_draft. This incidentally fixes a second latent issue: the sampling position was previously derived from a buffer clamped at max_model_len, so near the context limit two consecutive draft steps could reuse one noise vector. It is now tracked unclamped.

# Tests
```
gdelfin@gb200-rack1-18:~/vllm$ python -m pytest tests/v1/spec_decode/test_rejection_sampler_utils.py::test_gumbel_drafted_rejection_sample_is_unbiased -q
Running 2 items in this shard
..                                                                                                                                                                                                                                                                                                                          [100%]
2 passed in 2.96s

gdelfin@gb200-rack1-18:~/vllm$ python -m pytest tests/v1/worker/test_gpu_gumbel_sample.py::test_drafting_uses_a_separate_noise_stream -q
Running 1 items in this shard
.                                                                                                                                                                                                                                                                                                                           [100%]
1 passed in 3.92s
```
# Benchmarks
`vllm bench serve --dataset-name speed_bench --speed-bench-dataset-subset throughput_2k --speed-bench-output-len 2048`
— 2K in / 2K out, 512 prompts, 64 warmups, concurrency 64, `temperature=1.0 top_p=0.95 top_k=20`.
All runs use `draft_sample_method: probabilistic` (the salt is a no-op under the default `greedy`,
where the draft never calls `gumbel_sample`). 4× GB200, TP4 except Gemma 4 (TP1).
Baseline `de9250ac9e` → this PR `1529dd5363`.

| Model | N | Output tok/s | Acceptance length | Acceptance rate % | Median TTFT (ms) | Median TPOT (ms) | Median ITL (ms) |
|:--|--:|:--|:--|:--|:--|:--|:--|
| DeepSeek V4 Flash + MTP | 3 | 5444 → 5438 (-0.11%) | 2.589 → 2.587 (-0.09%) | 52.96 → 52.89 (-0.14%) | 221.0 → 226.1 (+2.30%) | 11.64 → 11.60 (-0.29%) | 20.11 → 20.11 (+0.01%) |
| DeepSeek V4 Flash + DSpark | 5 | 5825 → 5942 (+2.00%) | 3.514 → 3.500 (-0.39%) | 50.28 → 50.00 (-0.55%) | 237.1 → 225.9 (-4.74%) | 10.83 → 10.52 (-2.82%) | 24.05 → 23.99 (-0.23%) |
| MiMo v2.5 Pro + DFlash | 8 | 4111 → 4086 (-0.62%) | 3.794 → 3.771 (-0.61%) | 34.93 → 34.64 (-0.83%) | 210.2 → 206.7 (-1.68%) | 14.44 → 14.43 (-0.05%) | 44.14 → 44.15 (+0.03%) |
| Inkling NVFP4 + MTP | 8 | 3813 → 3864 (+1.33%) | 3.381 → 3.386 (+0.15%) | 29.76 → 29.82 (+0.22%) | 205.4 → 212.8 (+3.59%) | 16.22 → 16.07 (-0.92%) | 47.40 → 47.31 (-0.18%) |
| Gemma 4 E2B + MTP | 3 | 10609 → 10508 (-0.96%) | 2.623 → 2.612 (-0.39%) | 54.09 → 53.75 (-0.64%) | 92.5 → 94.5 (+2.18%) | 5.48 → 5.67 (+3.58%) | 6.80 → 6.69 (-1.54%) |

Per-position acceptance rate (%), baseline → this PR:

| Model | pos 0 | pos 1 | pos 2 | pos 3 | pos 4 | pos 5 | pos 6 | pos 7 |
|:--|--:|--:|--:|--:|--:|--:|--:|--:|
| DeepSeek V4 Flash + MTP | 85.2 → 85.1 | 53.1 → 53.1 | 20.6 → 20.5 | — | — | — | — | — |
| DeepSeek V4 Flash + DSpark | 83.6 → 83.5 | 64.6 → 64.4 | 47.3 → 46.9 | 33.1 → 32.8 | 22.8 → 22.5 | — | — | — |
| MiMo v2.5 Pro + DFlash | 76.5 → 76.3 | 56.1 → 55.9 | 42.2 → 41.9 | 32.5 → 32.3 | 25.5 → 25.1 | 19.9 → 19.5 | 15.4 → 15.1 | 11.4 → 11.0 |
| Inkling NVFP4 + MTP | 68.5 → 68.6 | 48.0 → 48.1 | 34.5 → 34.6 | 26.0 → 26.1 | 20.2 → 20.3 | 16.3 → 16.3 | 13.4 → 13.4 | 11.2 → 11.2 |
| Gemma 4 E2B + MTP | 70.8 → 70.5 | 51.8 → 51.4 | 39.7 → 39.3 | — | — | — | — | — |

**No regressions.** Throughput deltas straddle zero (-0.96% to +2.00%) and acceptance length is
within ±0.61% with both signs present — consistent with run-to-run variance rather than a
systematic effect. Per-position rates show no depth-dependent degradation: MiMo's 8-deep profile
moves uniformly by -0.2 to -0.4 pp with no trend across depth.

Acceptance is not expected to move: the acceptance test is `p(x) > u·q(x)`, whose expectation
depends only on `x ~ q`, which holds in both arms. An isolated probe with fixed `p`/`q` over 2M
independent trials put acceptance at 0.1068 (baseline) vs 0.1071 (this PR).

### CODE REVIEW

**Review by WoosukKwon — APPROVED** [review]
Thank you!
